File: gateOne/CreditCardCollector.java
import java.util.Scanner;

public class CreditCardCollector{
	public static void main(String [] args){
	
		Scanner Scanner = new Scanner(System.in);

		System.out.println("Hello, Kindly Enter Card Details to Verify");
		String CardNumber = Scanner.next();

		if(validOrInvalid(CardNumber))System.out.println("Invalid entry, check card number and retry");
		else {
			double totals  = AddTotalOfOddPlacesAndSecondDigitFromRightToLeft(CardNumber);
			String validity = ValidityStatusChecker(CardNumber);
			String type = getCategories(CardNumber);
			System.out.println("\n******************************");
			System.out.printf("**Credit Card Type: %s", type);
			System.out.printf("%n**Credit Card Number: %s", CardNumber);
			System.out.printf("%n**Credit Card Digit length: %s", CardNumber.length());
			System.out.printf("%n**Credit card Validity Status: %s",validity);
			System.out.println("\n******************************");
		}

	}
	
	public static boolean validOrInvalid(String CardNumber){
		return CardNumber.length() < 13 || CardNumber.length() > 16;
	}

	public static String getCategories(String CardNumber){
 		String category = "";
		if(CardNumber.charAt(0) == 4)category = "Visa Card";
		else if(CardNumber.charAt(0) == '5') category =  "Master Card";
		else if (CardNumber.charAt(0) == '6') category = "Discover Card";
		else if (CardNumber.charAt(0) == '3' && CardNumber.charAt(1) == '7') category = "American Express Card";
		else category = "Invalid";
		return category;
	}
	
	public static double getSecondDigitFromRightToLeft(String CardNumber){
		double total = 0;
		for (int index = CardNumber.length()-2; index >= 0; index-=2){
			int numbers = Integer.parseInt(String.valueOf(CardNumber.charAt(index)));
			numbers = numbers *2;
			if(numbers  > 9) numbers = numbers / 10 + numbers % 10;
				total = total + numbers;
				numbers = 0;
			}
		return total;
	}
	
	public static double getAllDigitsAtOddPlaces(String CardNumber){
		double totalOfOddPlaces = 0;

		for(int index = CardNumber.length()-1; index >= 0; index-=2){
			int numbers = Integer.parseInt(String.valueOf(CardNumber.charAt(index)));
			totalOfOddPlaces+= numbers;
			numbers	= 0;
		}
		return totalOfOddPlaces;
	}

	public static double AddTotalOfOddPlacesAndSecondDigitFromRightToLeft(String CardNumber){
		double sum = getAllDigitsAtOddPlaces(CardNumber) + getSecondDigitFromRightToLeft(CardNumber);
		return sum;
	}

	public static String ValidityStatusChecker(String CardNumber){
		String checker = "";
		if(AddTotalOfOddPlacesAndSecondDigitFromRightToLeft(CardNumber) % 10 == 0) checker = "Valid Card Number";
		else checker = "Invalid CardNumber";
		return checker;

	}




}	

File: gateOne/Credit card/CreditCardCollector.py
cardNumber = str(input("Hello, Kindly Enter Card Details To Verify: "))
cardNumberList = [int(digit) for digit in cardNumber]

if(len(cardNumberList) < 13 or len(cardNumberList) > 16):
	print('Invalid Card Length')
total = 0
sum_of_odd_places = 0

for digit in range (len(cardNumberList)-2,-1,-2):
	second_index = cardNumberList[digit] * 2 
	if(second_index > 9):
		second_index = second_index % 10 + second_index // 10
	total = total + second_index
#print(total)
for digit in range (len(cardNumberList)-1,0,-2):
	sum_of_odd_places = sum_of_odd_places + cardNumberList[digit] 
#print(sum_of_odd_places)
sum_total = sum_of_odd_places + total

print('*****************************************************')
if(cardNumberList[0] == 4):
	print('**Credit Card Type : Visa Card')
elif(cardNumberList[0] == 5):
	print('**Credit Card Type : Master Card')
elif(cardNumberList[0] == 6):
	print('**Credit Card Type : Discover Card')
elif(cardNumberList[0] == 3 and cardNumberList[1] == 7):
	print('**Credit Card Type :  American Express Card')
else:
	print('Card Type Not Recognised')
print('**Credit Card Number :' + cardNumber)
print('**Credit Card Digit Length :',len(cardNumber))

if(sum_total % 10 == 0):
	print('**Credit Card Validity Status :'+'valid')
else:
	print('**Credit Card Validity Status :'+'invalid')

print('*****************************************************')